dfastmi/batch/AnalyserDflowfm.py

```diff
@@ -0,0 +1,376 @@
+
+        if missing_data:
+            return missing_data, None
+
+        self._logger.log_load_mesh()
+        xn, yn, face_node_connectivity = self._get_xynode_connect(one_fm_filename)
+
+        xykm_data = self._get_xykm_data(xykm, xn, yn, face_node_connectivity)
+
+        if xykm is None and self._needs_tide:
+            self._logger.print_riverkm_needed_for_tidal()
+            return True, None
+
+        missing_data, dzq = self._get_dzq(discharges, rsigma, ucrit, filenames, n_fields, tide_bc, missing_data, xykm_data.iface, xykm_data.dxi, xykm_data.dyi)
+
+        if not missing_data:
+            self._logger.log_char_bed_changes()
+
+            if tstag > 0:
+                dzq = (dzq[0], dzq[0], dzq[1], dzq[2])
+                fraction_of_year = (fraction_of_year[0], tstag, fraction_of_year[1], fraction_of_year[2])
+                rsigma = (rsigma[0], 1.0, rsigma[1], rsigma[2])
+
+            # main_computation now returns new pointwise zmin and zmax
+            dzgemi, dzmaxi, dzmini, dzbi = main_computation(dzq, fraction_of_year, rsigma)
+
+            if self._old_zmin_zmax:
+                # get old zmax and zmin
+                dzmaxi = dzbi[0]
+                zmax_str = "maximum bed level change after flood without dredging"
+                dzmini = dzbi[1]
+                zmin_str = "minimum bed level change after low flow without dredging"
+            else:
+                zmax_str = "maximum value of bed level change without dredging"
+                zmin_str = "minimum value of bed level change without dredging"
+
+        sedimentation_data = None
+        if xykm is not None:
+            sedimentation_data = comp_sedimentation_volume(xykm_data.xni, xykm_data.yni, xykm_data.sni, xykm_data.nni, xykm_data.face_node_connectivity_index, dzgemi, slength, nwidth, xykm_data.xykline, self._outputdir, plotops)
+
+        return missing_data, OutputDataDflowfm(rsigma, one_fm_filename, xn, face_node_connectivity, dzq, dzgemi, dzmaxi, dzmini, dzbi, zmax_str, zmin_str, xykm_data, sedimentation_data)
+
+    def _get_first_fm_data_filename(self,q_threshold: float, discharges: Vector, rsigma: Vector, filenames: Dict[Any, Tuple[str,str]], tide_bc: Tuple[str, ...]) -> Tuple[str, bool]:
+        missing_data = False
+        one_fm_filename: Union[None, str] = None
+        # determine the name of the first FM data file that will be used
+        if 0 in filenames.keys(): # the keys are 0,1,2
+            one_fm_filename = self._get_first_fm_data_filename_based_on_numbered_keys(discharges, filenames)
+        else: # the keys are the conditions
+            one_fm_filename, missing_data = self._get_first_fm_data_filename_based_on_conditions_keys(discharges, filenames, tide_bc, rsigma, q_threshold)
+
+        if one_fm_filename is None:
+            self._logger.print_measure_not_active_for_checked_conditions()
+            missing_data = True
+
+        return one_fm_filename, missing_data
```

Comment:
I see missing_data a few times, can't this be a class field?

Reply:
I will look into  this.

Reply:
done, now a property